src/design-system/components/Box/Box.tsx

```diff
@@ -240,9 +264,19 @@ export const Box = forwardRef(function Box(
   return background ? (
     <BackgroundProvider color={background}>
       {backgroundStyle => (
-        <Component style={[backgroundStyle, ...style]} {...restProps} ref={ref}>
-          {children}
-        </Component>
+        <ApplyShadow
+          backgroundColor={backgroundStyle.backgroundColor}
+          enabled={Boolean(shadows)}
```

Comment:
Do we need this prop in addition to the `shadows` prop? Could this check be done internally to `ApplyShadow`?